middle_end/debuginfo.ml

```diff
@@ -73,6 +75,12 @@ let inline loc t =
   if loc == Location.none then t
   else (item_from_location loc) :: t
 
+let mark_as_tail_call t =
+  List.map (fun item -> { item with dinfo_is_tail_call = true; }) t
```

Comment:
Shouldn't this mark only the top location (which is the original location of the apply expression) as tail rather than all ?